File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        Map<String, String> logs = new TreeMap<>(new Comparator<String>() {
            @Override
            public int compare(String o1, String o2) {
                return o2.compareTo(o1);
            }
        });
        int N = Integer.parseInt(br.readLine());
        for (int i = 0; i < N; i++) {
            StringTokenizer tk = new StringTokenizer(br.readLine());
            String name = tk.nextToken();
            String type = tk.nextToken();
            if (type.equals("enter")) {
                logs.put(name, type);
            } else {
                if (logs.containsKey(name)) {
                    logs.remove(name);
                }
            }
        }
        for (String key : logs.keySet()) {
            bw.write(key + "\n");
        }
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/7785. 회사에 있는 사람/회사에 있는 사람.kt
import java.io.*
import java.util.*

fun main(args: Array<String>) = with(BufferedReader(InputStreamReader(System.`in`))){
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val s = mutableSetOf<String>()
    repeat(readLine().toInt()) {
        val token = StringTokenizer(readLine())
        val name = token.nextToken()
        if (s.contains(name)){
            s.remove(name)
        } else s.add(name)
        
    }
    val ss = s.sortedDescending()
    ss.forEach{
        bw.write("${it}\n")
    }
    bw.flush()
    bw.close()
}